python/taichi/snode/fields_builder.py

```diff
@@ -36,15 +36,26 @@ class FieldsBuilder:
     """
     _finalized_fbs = []
 
-    def __init__(self):
-        self._ptr = _snode_registry.create_root()
-        self._root = snode.SNode(self._ptr)
+    def __init__(self, ptr=None, root=None):
+        if ptr is None:
+            self._ptr = _snode_registry.create_root()
+        else:
+            self._ptr = ptr
+        if root is None:
+            self._root = snode.SNode(self._ptr)
+        else:
+            self._root = root
```

Comment:
```suggestion
    def __init__(self):
        self._ptr = _snode_registry.create_root()
        self._root = snode.SNode(self._ptr)
```

Reply:
I thought we can get root directly from cpp, instead of creating one in python?

Reply:
But it's not necessary to create a FieldsBuilder? I thought we keep a SNode is enough to achieve the effects we need.

Reply:
I mean `finalized_fbs` returns a list consists of `SNode`s, instead of a list consists of `FieldsBuilder`s.